docs/topics/multiplatform/multiplatform-discover-project.md

```diff
@@ -1,169 +1,315 @@
+    java.io.File("greeting.txt").writeText("Hello, Multiplatform!")
+}
+```
 
-        }
-        jsTest {
+As a solution, Kotlin creates platform-specific source sets, also referred to as platform source sets. Each target has a
+corresponding platform source set that compiles for only that target. For example, a `jvm` target has the
+corresponding `jvmMain` source set that compiles to only the JVM. Kotlin allows using platform-specific dependencies in
+these source sets, for instance, JDK in `jvmMain`:
 
-        }
-        nativeMain {
+```kotlin
+// jvmMain/kotlin/jvm.kt
+// You can use Java dependencies in the `jvmMain` source set
+fun jvmGreeting() {
+    java.io.File("greeting.txt").writeText("Hello, Multiplatform!")
+}
+```
 
-        }
-        nativeTest {
+### Compilation to a specific target
 
-        }
-    }
+Compilation to a specific target works with multiple source sets. When Kotlin compiles a multiplatform project to a
+specific target, it collects all source sets labeled with this target and produces binaries from them.
+
+Consider an example with `jvm`, `iosArm64`, and `js` targets. Kotlin creates the `commonMain` source set for common code
+and the corresponding `jvmMain`, `iosArm64Main`, and `jsMain` source sets for specific targets:
+
+![Compilation to a specific target](specific-target-diagram.svg){width=700}
+
+During compilation to the JVM, Kotlin selects all source sets labeled with "JVM", namely, `jvmMain` and `commonMain`. It
+then compiles them together to the JVM class files:
+
+![Compilation to JVM](compilation-jvm-diagram.svg){width=700}
+
+Because Kotlin compiles `commonMain` and `jvmMain` together, the resulting binaries contain declarations from
+both `commonMain` and `jvmMain`. This is how you can choose which parts of your application should be shared.
+
+When working with multiplatform projects, remember:
+
+* If you want Kotlin to compile your code to a specific platform, declare a corresponding target.
+* To choose a directory or source file to store the code, first decide among which targets you want to share your code:
+
+    * If the code is shared among all targets, it should be declared in `commonMain`.
+    * If the code is used for only one target, it should be defined in a specific source set for that target (for
+      example, `jvmMain` for the JVM).
+* Code written in platform-specific source sets can access declarations from the common source set. For example, the
+  code in `jvmMain` can use code from `commonMain`. However, the opposite isn't true: `commonMain` can't use code
+  from `jvmMain`.
+* Code written in platform-specific source sets can use the corresponding platform dependencies. For example, the code
+  in `jvmMain` can use Java-only libraries, like [Guava](https://github.com/google/guava)
+  or [Spring](https://spring.io/).
+
+### Intermediate source sets
+
+In simple multiplatform projects, only common code is shared among all available platforms and platform-specific code.
```

Comment:
The meaning is significantly distorted after the rephrasing. What the text wanted to say is that there's a specific class of multiplatform project ("simple multiplatform projects"), and in such projects, there's only "common code" and "platform-specific code"

Attempt to fix it back:

> Simple multiplatform projects usually have only common code and platform-specific code. The common code is represented by `commonMain` source set and shared among all declared targets. Platform-specific code is represented by platform-specific source sets like `jvmMain` and compiled only to that one respective target

Last two sentences can be removed if it's too verbose, leaving only:

> Simple multiplatform projects usually have only common code and platform-specific code.